mono/metadata/gc.c

```diff
@@ -726,10 +726,18 @@ finalize_domain_objects (DomainFinalizationReq *req)
 #elif defined(HAVE_SGEN_GC)
 	while (!suspend_finalizers && (count = mono_gc_finalizers_for_domain (domain, to_finalize, NUM_FOBJECTS))) {
```

Comment:
We should remove mono_gc_finalizers_for_domain from the sgen api and instead have a function that does all that queueing instead.